app/services/decisions/case_decision_tree.rb

```diff
@@ -101,22 +101,20 @@ def after_hearing_details
     # NOTE: for MVP, this is the last step of the application,
     # however post-MVP there will be means assessment steps,
     # unless the applicant has been passported. As this is not
-    # yet implemented, and for MVP we only let means passported
-    # applicants, raise an exception (this should NOT happen but
-    # in case it happens, we get alerted).
+    # yet implemented, for now both branches behave the same.
+    # However before submission we perform a validation to
+    # ensure applicant is means-passported (array is not empty).
     #
-    # rubocop:disable Style/GuardClause
+    # rubocop:disable Style/IdenticalConditionalBranches,Lint/DuplicateBranch
     def after_ioj
       if Passporting::MeansPassporter.new(current_crime_application).call
         edit('/steps/submission/review')
       else
         # TODO: post-MVP implement means assessment steps
-        # For MVP, this branch should not be reachable, and if
-        # we end up here, we want to know about it (Sentry)
-        raise InvalidStep, 'application is not means-passported'
```

Comment:
I've removed the raised exception here, as we now want to catch the means passport problem with the new "pre-submission" validation, and present an error to the user to hopefully recover.
I've left on purpose the 2 branches even if they go to the same place for clarity and is silly.